### PR TITLE
[benchmark] add gpt benchmark

### PR DESCRIPTION
**Pull request by ver217**

### CODE REVIEW

File: benchmark/gpt/callback.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import psutil
+import torch
+import torch.distributed as dist
+from pytorch_lightning.callbacks import Callback
+
+
+def print_rank_0(*args, **kwargs):
+    if dist.get_rank() == 0:
+        print(*args, **kwargs)
```
Comment (1SAA) [inline benchmark/gpt/callback.py:9]: Maybe adding `dist.barrier()`  in the end of this function makes the function more robust.

File: benchmark/gpt/callback.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def __init__(self) -> None:
+        super().__init__()
+        self.max_cpu_mem = 0
+
+    def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx) -> None:
+        self.max_cpu_mem = max(get_cpu_mem(), self.max_cpu_mem)
+
+    def on_fit_start(self, trainer, pl_module) -> None:
+        max_cuda_mem = torch.cuda.max_memory_allocated()
+        print_rank_0(f'CPU memory before training: {get_cpu_mem()/1024**2:.3f} MB')
```
Comment (1SAA) [inline benchmark/gpt/callback.py]: Is it better to use `torch.cuda.memory_allocated()` in the start?
Maybe display them both is better?

Comment (ver217) [reply benchmark/gpt/callback.py]: done

File: benchmark/gpt/callback.py (hunk 1 above)
Comment (ver217) [reply benchmark/gpt/callback.py:9]: done

**Review by 1SAA — APPROVED** [review]

**Pull request merged**